### PR TITLE
COMPAT: Make is_number import backwards-compat

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

xref #304.

### CODE REVIEW

File: pandas_datareader/edgar.py [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 from pandas.compat import StringIO
-from pandas.api.types import is_number
+from distutils.version import LooseVersion
+from pandas import read_csv, DataFrame, to_datetime
 
 from pandas_datareader.base import _BaseReader
 from pandas_datareader.compat import BytesIO
 from pandas_datareader._utils import RemoteDataError
 
+if LooseVersion(pd.__version__) <= '0.19.2':
```
Comment (jreback) [inline pandas_datareader/edgar.py]: would put this in ``pandas_datareader/compat/__init__.py``

```
if LooseVersion(.....):
       from pandas.core import common as pdapi
else:
      from pandas.api import types as pdapi
```

then you can 

``from pandas_datareader.compat import pdapi``

Comment (gfyoung) [inline pandas_datareader/edgar.py]: That works too.

Comment (jreback) [inline pandas_datareader/edgar.py]: if directly import ``is_number`` in compat is ok too. this then isolates all version specific things there.

Comment (gfyoung) [inline pandas_datareader/edgar.py]: That works too.

**Comment by jreback** [comment]
thanks!

**Pull request merged**